sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -81,6 +81,11 @@ def _f(value):
 
   return _f
 
+def get_all_subclasses(cls):
+    """Returns all subclasses of a class, recursively."""
+    return set(cls.__subclasses__()).union(
+        [s for c in cls.__subclasses__() for s in get_all_subclasses(c)])
```

Comment:
It looks like this is causing a bunch of test failures because its not correctly deduping items and it considers the current class multiple times. I'm guessing the set comparison isn't working correctly. Could you please take a look?